### PR TITLE
feat: IdentifyRecurringTransactionsStage (#163)

### LINKED ISSUE #163: Pipeline stage: Identify recurring transactions
## Description

Implement the `IdentifyRecurringTransactionsStage` — the most broadly useful stage that runs on every sync. It analyzes imported Basiq transactions to find repeating patterns (same description, same amount, consistent frequency) and creates suggestions for the user to confirm.

**Parent Epic:** #160
**Depends on:** #162 (Orchestrator)

## Algorithm

**File:** `app/Services/PipelineStages/IdentifyRecurringTransactionsStage.php`

- **key:** `'identify-recurring-transactions'`
- **shouldRun:** Always `true` (runs every sync)

### Detection Logic

1. Load current Basiq-sourced transactions where `planned_transaction_id IS NULL`
2. **Normalize descriptions:** prefer `merchant_name` > `clean_description` > strip trailing noise from `description` (card numbers like 4+ digits, date patterns DD/MM, country codes)
3. Group by: normalized description + direction + account_id
4. For each group with **2+ members:**
   - Amount consistency: all amounts within 5% of median
   - Calculate intervals between consecutive transactions (sorted by `post_date`)
   - Map median interval to `RecurrenceFrequency` enum (5-9d=Weekly, 12-16d=Fortnightly, 27-35d=Monthly, etc.)
   - Calculate confidence score based on: match count, amount consistency, interval consistency
5. Create `AnalysisSuggestion` records (type: `RecurringTransaction`)

### Suggestion Payload Schema

```json
{
  "description": "NETFLIX",
  "clean_description": "Netflix",
  "amount": 2299,
  "direction": "debit",
  "frequency": "every-month",
  "account_id": 42,
  "category_id": null,
  "matched_transaction_ids": [501, 602, 703],
  "start_date": "2026-01-15",
  "confidence_score": 0.95
}
```

### Idempotency Rules

- Skip if existing pending/accepted suggestion matches description + account_id
- Skip if `PlannedTransaction` already exists with matching description + amount (±5%) + frequency + account
- Skip if rejected suggestion with same criteria exists within last 90 days

### Description Normalization Examples

```
"WOOLWORTHS 1234 SYDNEY AU" → "WOOLWORTHS"
"NETFLIX.COM NETFLIX.COM" → "NETFLIX.COM"
"BP DANDENONG 12/03" → "BP DANDENONG"
```

Strip trailing: card numbers (4+ digits), dates (DD/MM, MM/DD), country codes (2-letter at end), location suffixes. This is heuristic — the user confirms via suggestion.

## Tasks

- [ ] Create `IdentifyRecurringTransactionsStage` implementing `PipelineStageContract`
- [ ] Implement description normalization (private method or small utility)
- [ ] Implement frequency detection from interval analysis
- [ ] Implement idempotency checks (existing suggestions + existing PlannedTransactions + rejected memory)
- [ ] Implement confidence scoring
- [ ] Write comprehensive tests with factory-generated transaction data:
  - Monthly recurring (rent, subscriptions)
  - Weekly recurring (grocery shopping)
  - Fortnightly recurring (loan repayments)
  - Edge cases: varying descriptions, slight amount differences
  - Idempotency: re-running doesn't duplicate
  - Rejected memory: previously rejected suggestions not re-suggested
- [ ] Run `op ci` to verify

## Acceptance Criteria

- Detects monthly, weekly, fortnightly recurring patterns
- Description normalization handles common Australian bank transaction formats
- Idempotent — re-running produces no duplicates
- Respects rejected suggestions for 90 days
- Creates audit entries for each suggestion created and each skip
- Confidence scoring distinguishes strong vs weak matches
- All tests pass, `op ci` green

### PR DESCRIPTION
**Pull request by robwilde**
## Summary

- Implements the first concrete pipeline stage: `IdentifyRecurringTransactionsStage`, which detects recurring transaction patterns from imported Basiq bank data
- Groups unmatched Basiq transactions by normalized merchant/description, direction, and account, then analyzes amount consistency, interval regularity, and calculates a confidence score to create `AnalysisSuggestion` records (type: `RecurringTransaction`)
- Includes idempotency checks — skips groups that already have an accepted suggestion, a matching active `PlannedTransaction`, or a rejection within the last 90 days
- Registers the stage in the `TransactionAnalysisPipeline` singleton in `AppServiceProvider`

## Changes

| File | Change |
|------|--------|
| `app/Services/PipelineStages/IdentifyRecurringTransactionsStage.php` | New stage class (~390 lines) implementing `PipelineStageContract` |
| `tests/Feature/Services/PipelineStages/IdentifyRecurringTransactionsStageTest.php` | 33 Pest feature tests (~700 lines) covering detection, normalization, amount consistency, idempotency, edge cases, confidence scoring, payload shape, audit trail, and integration |
| `app/Providers/AppServiceProvider.php` | Registers stage in pipeline singleton (stages: [] → stages: [new IdentifyRecurringTransactionsStage]) |

## Detection Logic

- **Normalization**: Prefers `merchant_name` > `clean_description` > cleaned raw `description` (strips card numbers, trailing dates, country codes, deduplicates repeated phrases)
- **Frequency mapping**: Median interval → `RecurrenceFrequency` enum (weekly through yearly, with real-world tolerance ranges)
- **Confidence formula**: `0.30 × matchCount + 0.35 × amountConsistency + 0.35 × intervalRegularity` (minimum threshold: 0.40)
- **Idempotency**: Checks accepted suggestions, matching PlannedTransactions (5% amount tolerance), and recent rejections (90-day cooldown)

## Test plan

- [x] `op test.filter IdentifyRecurringTransactionsStageTest` — 33 tests, 81 assertions pass
- [x] `op test.filter TransactionAnalysisPipelineTest` — 19 tests, 49 assertions pass (no regression)
- [x] `op ci` — full suite passes (1122 tests, 0 PHPStan errors, Pint clean)

Closes #163

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/Services/PipelineStages/IdentifyRecurringTransactionsStage.php [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+        foreach ($groups as $group) {
+            if ($group->count() < 2) {
+                continue;
+            }
+
+            $analysis = $this->analyzeGroup($group);
+
+            if ($analysis === null) {
+                continue;
+            }
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Services/PipelineStages/IdentifyRecurringTransactionsStage.php:78]: Intentional by design — only idempotency skips are audited. Analysis rejections (inconsistent amounts, unmappable intervals, low confidence) are the normal case for most transaction groups and would create noise in the audit table without actionable value. The audit trail here answers the specific operational question: "why didn't the pipeline suggest X when the pattern exists?" — that's exclusively an idempotency concern.

File: app/Services/PipelineStages/IdentifyRecurringTransactionsStage.php [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+    }
+
+    private function hasMatchingPlannedTransaction(User $user, string $description, int $accountId, RecurrenceFrequency $frequency, int $medianAmount): bool
+    {
+        return PlannedTransaction::query()
+            ->where('user_id', $user->id)
+            ->where('account_id', $accountId)
+            ->where('frequency', $frequency)
+            ->where('is_active', true)
+            ->whereRaw('UPPER(description) = ?', [$description])
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Services/PipelineStages/IdentifyRecurringTransactionsStage.php:318]: The `whereLike` suggestion subtly changes semantics — `LIKE` interprets `%` and `_` as wildcards, so it's not a drop-in replacement for exact equality matching. A PlannedTransaction description containing those characters (unlikely but possible) would match more broadly than intended.

The `whereRaw('UPPER(description) = ?')` uses strict equality and makes the case-insensitive intent explicit. `UPPER()` is standard SQL supported by MySQL, PostgreSQL, and SQLite, so portability is not a practical concern here. Keeping as-is.
